src/data/getDataBy.ts

```diff
@@ -1,5 +1,30 @@
 import {ensureArray} from 'utilities'
 
+interface ActionComboData {
+	start?: boolean
+	from?: number
+	end?: boolean
+}
+
+export interface ActionData {
+	id: number
+	name: string
+	icon: string
+	onGcd?: boolean
+	breaksCombo?: boolean
+	combo?: ActionComboData
+	castTime?: number
+	cooldown?: number
+	autoAttack?: boolean
+}
```

Comment:
purely because action data's been heavily over-used as meta information, can i just get a

```ts
[key: string]: unknown
```

added to the end of this action interface?

It'll permit people to use custom keys in the action data without editing core, but the `unknown` will force them to typecast it before usage (so it's easier to see mistakes when reviewing)